models/UserManager.php

```diff
@@ -0,0 +1,289 @@
+
+            return $exist;
+        }else{
+            return 0;
+        }
+    }
+
+    // Fonction qui vérifie qu'un utilisateur est bien administrateur
+    function is_admin($email,$password){
+        $tableau = [
+            'email'     =>  $email,
+            'password'  =>  sha1($password)
+        ];
+        $sql = "SELECT * FROM admins WHERE email = :email AND password = :password";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+        $exist = $req->rowCount($sql);
+        
+        return $exist;
+    }
+
+    function hasnt_password(){
+        $sql = "SELECT * FROM admins WHERE email = '{$_SESSION['admin']}' AND password = ''";
+        $req = $this->db->prepare($sql);
+        $req->execute();
+        $exist = $req->rowCount($sql);
+        
+        return $exist;
+    }
+
+    // Fonction qui gère les settings
+    function email_taken($email){
+        $tableau = ['email'   =>  $email];
+        $sql = "SELECT * FROM admins WHERE email = :email";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+        $free = $req->rowCount($sql);
+
+        return $free;
+    }
+
+    //Fonction permettant de créer un token
+    function token($length){
+        $chars = "azertyuiopqsdfghjklmwxcvbnAZERTYUIOPQSDFGHJKLMWXCVBN0123456789";
+        
+        return substr(str_shuffle(str_repeat($chars,$length)),0,$length);
+    }
+
+    // Fonction permettant d'ajouter un modérateur / administrateur et de lui envoyer un mail
+    function add_modo($name,$email,$role,$token){
+        $tableau= [
+            'name'      =>  $name,
+            'email'     =>  $email,
+            'token'     =>  $token,
+            'role'      =>  $role
+        ];
+
+        $sql = "INSERT INTO admins(name,email,token,role) VALUES(:name,:email,:token,:role)";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+
+        $subject = "Modo / Admin sur le blog";
+        $message = '              
+           <html lang="en" style="font-family: sans-serif;">
+                <head>
+                    <meta charset="UTF-8">
+                </head>
+                <body>
+                    Voici votre identifiant et code unique à insérer sur <a href="http://projet-5.jeromelacquemant.fr/index.php?page=new">cette page</a>.
+                    <br/>Identifiant: '.$email.'
+                    <br/>Mot de passe: '.$token.'
+                    <br/>Vous êtes: '.$role.'
+                    <br/><br/>Après avoir inséré ces informations, vous devrez choisir un mot de passe.
+                </body>
+            </html>
+        ';
+
+        $header = "MIME-Version: 1.0\r\n";
+        $header .= "Content-type: text/html; charset=UTF-8\r\n";
+        $header .= 'From: jerome.lacquemant@gmail.com' . "\r\n" . 'Reply-To: jerome.lacquemant@gmail.com' . "\r\n" . 'X-Mailer: PHP/' . phpversion();
+
+        mail($email,$subject,$message,$header);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function mail() is discouraged](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)